src/widget/live.js

```diff
@@ -51,45 +47,33 @@ const Widget = function(el) {
 }
 
 const initializeWidget = function(input, targetClass) {
-  const inputArr = $(input)
-
-  if (inputArr.length === 0) {
-    throw new Error('No DOM elements found matching selector')
-  } else if (inputArr.length > 1) {
-    warn('There are multiple DOM elements matching selector')
-  }
-
-  input = inputArr.eq(0)
-
-  const s = build(input.data())
+  const s = build(input.dataset)
   const Widget = s.multiple ? MultipleWidgetClass : WidgetClass
 
   if (targetClass && Widget !== targetClass) {
     throw new Error(`This element should be processed using ${Widget._name}`)
   }
 
-  let api = input.data(dataAttr)
-  if (!api || api.inputElement !== input[0]) {
+  let api = cache.get(input)
+  if (!api || api.inputElement !== input) {
+    cache.delete(input)
     cleanup(input)
     const widget = new Widget(input, s)
 
     api = widget.api()
-    input.data(dataAttr, api)
+    input.dataset[dataAttr] = api
```

Comment:
I think that we need to put this not in input but in the collection (Map or WeakMap)

Reply:
fixed